tests/utils.go

```diff
@@ -419,7 +420,7 @@ func validateRepoConfig(t *testing.T, repoKey string, params interface{}) bool {
 		return false
 	}
 	for key, value := range paramsMap {
-		if confMap[key] != value {
+		if !reflect.DeepEqual(confMap[key], value) {
 			t.Error(fmt.Sprintf("Key %s: Expected: %v, Actual: %v", key, value, confMap[key]))
 			return false
 		}
```

Comment:
Reflection is forbidden, unless we have no choice.
I suggest remove the bool return value from this function and replace all `if -> error` with assertions.
For example: 
```go
for key, value := range paramsMap {
    assert.NotEqual(t, confMap[key], value)
}
```